tests/unit-test/test_serializer.c

```diff
@@ -244,34 +244,37 @@ void test_recv_mam_message_request_ntru_deserialize(void) {
   ta_recv_mam_req_t* req = recv_mam_req_new();
 
   TEST_ASSERT_EQUAL_INT32(SC_OK, recv_mam_message_req_deserialize(json, req));
-  data_id_mam_v1_t* data_id = (data_id_mam_v1_t*)req->data_id;
+  recv_mam_data_id_mam_v1_t* data_id = (recv_mam_data_id_mam_v1_t*)req->data_id;
   TEST_ASSERT_EQUAL_STRING(TEST_BUNDLE_HASH, data_id->bundle_hash);
   TEST_ASSERT_NULL(data_id->chid);
   TEST_ASSERT_NULL(data_id->epid);
   TEST_ASSERT_NULL(data_id->msg_id);
 
-  key_mam_v1_t* key = (key_mam_v1_t*)req->key;
+  recv_mam_key_mam_v1_t* key = (recv_mam_key_mam_v1_t*)req->key;
   TEST_ASSERT_EQUAL_MEMORY(TEST_NTRU_PK, key->enc_key, NUM_TRYTES_MAM_NTRU_PK_SIZE);
 
   recv_mam_req_free(&req);
 }
 
 void test_send_mam_message_request_deserialize(void) {
   const char* json =
-      "{\"seed\":\"" TRYTES_81_1
-      "\",\"channel_ord\":" STR(TEST_CHANNEL_ORD) ",\"message\":\"" TEST_PAYLOAD "\",\"ch_mss_depth\":" STR(
-          TEST_CH_DEPTH) ",\"ep_mss_depth\":" STR(TEST_EP_DEPTH) ",\"ntru_pk\":\"" TEST_NTRU_PK
-                                                                 "\",\"psk\":\"" TRYTES_81_2 "\"}";
-  ta_send_mam_req_t* req = send_mam_req_new();
+      "{\"x-api-key\":\"" TEST_TOKEN "\",\"data\":{\"seed\":\"" TRYTES_81_1 "\",\"message\":\"" TEST_PAYLOAD
+      "\",\"ch_mss_depth\":" STR(TEST_CH_DEPTH) ",\"ep_mss_depth\":" STR(
+          TEST_EP_DEPTH) "},\"key\":{\"ntru\":[\"" TEST_NTRU_PK "\"],\"psk\":[\"" TRYTES_81_2 "\",\"" TRYTES_81_3
```

Comment:
Should the payload contain the string `protocol: MAM_V1`?

Reply:
protocol is for TA to process the request only, so I think I can just put it in another field in request.